multi-educator/src/Dscp/MultiEducator/Web/Server.hs

```diff
@@ -68,45 +66,25 @@ mkMultiEducatorApiServer _aaaConfig nat =
         nat
         (\eData -> mkEducatorApiServer' $ eadId eData)
 
-mkStudentApiServer'
-    :: forall ctx m. MultiEducatorWorkMode ctx m
-    => (forall x. m x -> Handler x)
-    -> EducatorBotConfigRec
-    -> m (Text -> Server ProtectedStudentAPI)
-mkStudentApiServer' nat botConfig = do
-    case botConfig ^. tree #params . selection of
-      _ {-EducatorBotOff-} -> return $ \login -> getServer login . studentApiHandlers
-      -- To initilialize the bot we need a database, how do we choose a database?
-      {-EducatorBotOn params -> initializeBot params $ do
-        return $ (\student -> getServer . addBotHandlers student . studentApiHandlers $ student)-}
-  where
-    getServer login handlers = hoistServerWithContext
-        rawStudentAPI
-        (Proxy :: Proxy '[StudentCheckAction])
-        (\x -> nat $ lookupEducator login >>= \c -> normalToMulti c x)
-        (toServant handlers)
-
 mkStudentApiServer
     :: forall ctx m. MultiEducatorWorkMode ctx m
     => (forall x. m x -> Handler x)
-    -> EducatorBotConfigRec
     -> m (Server MultiStudentAPI)
-mkStudentApiServer nat botParams = do
-    studApi <- mkStudentApiServer' nat botParams
-    return $ hoistServerWithContext
-        multiStudentAPI
-        (Proxy :: Proxy '[StudentCheckAction, NoAuthContext "student"])
-        id
-        studApi
+mkStudentApiServer nat =
+    return $ \login student ->
+        hoistServerWithContext
+            rawStudentAPI
+            (Proxy :: Proxy '[StudentCheckAction])
+            (\x -> nat $ lookupEducator login >>= \c -> normalToMulti c x)
+            (toServant $ studentApiHandlers student)
 
 -- | Create an action which checks whether or not the
 -- student is a valid API user.
 -- If bot is enabled, all students are allowed to use API.
 createStudentCheckAction
     :: forall ctx m. MultiEducatorWorkMode ctx m
-    => EducatorBotConfigRec
-    -> m StudentCheckAction
-createStudentCheckAction _ =
+    => m StudentCheckAction
+createStudentCheckAction =
     return . StudentCheckAction . const $ pure True
 {-
 createStudentCheckAction EducatorBotOff = do
```

Comment:
You missed a piece of bot comments here

Reply:
Hm, they are not fully bot-related, probably this commented check should happen in real life but I do not enable it because it would make testing more difficult :smirk_cat: 
@flyingleafe what do you think?

Anyway, I will remove that `EducatorBotOff` argument, thanks for the notice.